ivy/functional/frontends/torch/nn/functional/loss_functions.py

```diff
@@ -208,43 +208,80 @@ def l1_loss(
 
 @to_ivy_arrays_and_back
 def nll_loss(
-    input,
-    target,
-    weight=None,
-    size_average=None,
-    ignore_index=-100,
-    reduce=None,
-    reduction="mean",
+        input,
+        target,
+        weight=None,
+        size_average=None,
+        ignore_index=-100,
+        reduce=None,
+        reduction="mean",
 ):
-
     out = ivy.zeros_like(target)
-
+    
     if len(input.shape) == 1:
         for i in range(len(target)):
             out[i] = input[target[i]]
     else:
         for i in range(len(target)):
             out[i] = input[i][target[i]]
     loss = -out
-
+    
     if weight is not None:
         loss = ivy.multiply(weight, loss)
     reduct = _get_reduction(reduction, size_average, reduce)
     ret = reduct(loss)
-
+    
     return ret
 
 
 @to_ivy_arrays_and_back
 @with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
 def soft_margin_loss(
-    input,
-    target,
-    size_average=None,
-    reduce=None,
-    reduction="mean",
+        input,
+        target,
+        size_average=None,
+        reduce=None,
+        reduction="mean",
 ):
     loss = ivy.log1p(ivy.exp(-input * target))
     reduction = _get_reduction(reduction, size_average, reduce)
     ret = reduction(loss)
     return ret
+
+
+@to_ivy_arrays_and_back
+def kl_div(
+        input,
+        target,
+        size_average=None,
+        reduce=None,
+        reduction='mean',
+        log_target=False
+):
+    size = ivy.shape(input)
+    
+    def loss_fn():
+        if log_target:
+            return ivy.exp(target) * (target - input)
+        return target * (ivy.log(target) - input)
+    
+    def batchmean(x):
+        if not reduce:
+            return x / size[0]
```

Comment:
this should accommodate for cases where arrays are 0-dim.

when an array is 0-dim, the shape is a tuple of (), meaning length 0. the test fails in this case

Reply:
I've fixed this by adding a handler for the size length.

Reply:
I also updated the test case to handle scalar now.